### DISCUSSION TITLE
Event Modifier On QSelect @remove Event

### DISCUSSION THREAD

**Discussion by mecjos** [quasarframework/quasar#12659] [General - Components / Directives / etc] [answered]
Hi, I want to remove an option from selection conditionally.. I tried to use @remove event but I couldn't stop remove event before my check... I want to do something below;

```
<q-select @remove.stop="isItRemovable"></q-select>
.
.
.
methods: {
  isItRemovable(option) {
    if (option.value.name === 'yes removable') {
        remove
    } else {
       show not removable warning
    }
  }
}
```

how can i do this? thank you.

**Comment by MilosPaunovic** [comment]
Have you tried something in the lines of conditional expression, like:

```js
<q-select
  @remove="(details) => details.isRemovable ? callRemoveFunction() : false"
  ...otherProps
</q-select>
```

**Reply by mecjos** [reply]
this doesn't work..

**Reply by MilosPaunovic** [reply]
What exactly does not work?
`details.isRemovable` is something you need to change to condition you want to check if the tag is removable.

**Reply by mecjos** [reply]
it removes again even in false condition... I tried... if you succeed it, could you send me a codepen link please?

**Reply by MilosPaunovic** [reply]
I made a mistake, it's not the way to handle it. I apologize.

Right now, I can't think any other solution to this issue other than handling the logic for adding/removing chips yourself, via slots. Maybe somebody else can think of something better.

**Comment by yusufkandemir** [comment, marked as answer]
https://vuejs.org/guide/essentials/event-handling.html#event-modifiers
`.stop` modifier is basically `event.stopPropagation()`, and using it will always call it. So, it's definitely not what you are thinking or what you should you use. If you want to prevent an event from what it should do, e.g. `@submit` of a native `form` reloading the page, you would use the `.prevent` modifier. However, it's still not what you want, because using it will always prevent it, not conditionally. Also, `@remove` is not a native event, so you can't prevent its default action. So, calling `event.preventDefault()` manually on condition wouldn't also work.

`@remove` is called after the option is removed from the selection, so at the point where your callbacks get called, everything has already happened. So, you can't prevent something that is already happened. It is there to just let you know it happened.

You could try adding that option back into the selection after it got removed if it wasn't supposed to be removed. But, this may cause problems and might not be really clean.

If you want to actually prevent the remove action before it happens, you would need to take control of the places that can trigger it. So, it would be clicking an already selected option on the dropdown(_when using `multiple` prop_), when clicking X icons on chips(_`use-chips` prop_), when clicking the X icon on the component itself(_`clearable` prop_), and any other custom logic you have that can initiate a removal. So, as those props don't have a feature such as disabling the removal as you want, you would need to implement those features yourself to have full control over them.

- If you were using `multiple` prop and want to disallow some options from getting removed after re-clicking them on the options menu, then you have mainly two options. You can use `option-disable` prop, and structure in a way that an option gets disabled if they are selected and meet a certain condition. Or, you can use `option` slot to render the options yourself, and prevent it from getting clicked if it meets a condition.
- If you were using `use-chips`, you now need to use `selected` or `selected-item` slots and handle the removal yourself. You can also just disable the removal icons instead of showing a warning when clicked.
- If you were using `clearable` prop, you now need to use `append` slot to display the clear icon, handle it yourself, disallow the clear all operation if some options meet some conditions, etc.

**Reply by mecjos** [reply]
Thank you for this explanatory answer Yusuf. . I think option-disable method will be ok for me...

**Comment by pdanpdan** [comment]
The model change for QSelect is done using the `@update:model-value` event.
`@add` and `@remove` are not by default used to update the model and are emitted before `@update:model-value` is emitted.

So you can:
- use `:model-value="x" @update:model-value="myFunctionThatChecksTheChangesAndUpdatesX"` and in that function check the existing model and the new model (param of the function) and update the model as you see fit
- use `:model-value="x" @add="addToX" @remove="removeFromX"` and only change the model if you want
